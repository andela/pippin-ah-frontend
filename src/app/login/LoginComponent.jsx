import React, { Fragment } from 'react';
import { Redirect } from 'react-router-dom';
import { RingLoaderComponent } from '../loaders';
import { constants } from './duck';
import './signin.scss';

export const LoginComponent = ({ loginUser, loginState }) => {
  const onFormSubmit = e => {
    e.preventDefault();
    const usernameOrEmail = e.target.elements.usernameOrEmail.value.trim();
    const password = e.target.elements.password.value.trim();
    loginUser(usernameOrEmail, password);
  };
  if (loginState === constants.LOGIN_SUCCESS) {
    return <Redirect to="/" />;
  }
  return (
    <Fragment>
      <div className="relative-div-signIn">
        <div className="signIn-div-wrapper">
          <div className="signIn-container">
            <div className="close-button-div-signIn">
              <p className="close-button-sign close"> X </p>
            </div>
            <div className="signIn-text">
              Sign <span className="up">In</span>
            </div>
            <div className="signIn-wrapper">
              <div className="social-media-icons-signIn">
                <div className="btn-facebookk">
                  <div className="facebook-icon-div-signIn">
                    <i className="fab fa-facebook-f" />
                  </div>
                  <div className="fb-text-wrapper-signIn">
                    <span className="social-text-signIn">
                      Sign in with Facebook
                    </span>
                  </div>
                </div>
                <div className="btn-twitterr">
                  <div className="twitter-icon-div-signIn">
                    <i className="fab fa-twitter" />
                  </div>
                  <div className="twitter-text-wrapper-signIn">
                    <span className="social-text-signIn">
                      Sign in with Twitter
                    </span>
                  </div>
                </div>
                <div className="btn-googlee">
                  <div className="google-icon-div-signIn">
                    <i className="fab fa-google" />
                  </div>
                  <div className="google-text-wrapper-signIn">
                    <span className="social-text-signIn">
                      Sign in with Google
                    </span>
                  </div>
                </div>
              </div>
              <form className="login" onSubmit={onFormSubmit}>
                <div className="input-groups-signIn">
                  <div className="form-icon-div-signIn">
                    <i className="fas fa-envelope form-input-icons-signIn" />
                  </div>
                  <input
                    id="usernameOrEmail"
                    type="text"
                    name="usernameOrEmail"
                    placeholder="Username or Email"
                    required
                  />
                </div>
                <div className="input-groups-signIn">
                  <div className="form-icon-div-signIn">
                    <i className="fas fa-key form-input-icons-signIn" />
                  </div>
                  <input
                    id="password"
                    type="password"
                    name="password"
                    placeholder="Password"
                    required
                  />
                </div>
                <button className="btn-submit-signIn" type="submit">
                  SIGN IN
                </button>
                <div className="sign-up-alternative-div-signIn">
                  <p className="sign-up-alternative-text-signIn">
                    {' '}
                    Forgort Password?
                  </p>
                </div>
              </form>
            </div>
            <div className="ring-loader">
              {loginState === constants.LOGGING_IN && <RingLoaderComponent />}
            </div>
          </div>
        </div>
      </div>
    </Fragment>
  );
};

export default LoginComponent;
