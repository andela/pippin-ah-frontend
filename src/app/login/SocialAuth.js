import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { socialMediaLogin } from './duck/operations';
import constants from './duck/constants';

/* eslint-disable react/prefer-stateless-function */
class SocialAuth extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    const { dispatch } = this.props;
    console.log(window.location);
    dispatch(socialMediaLogin(window.location.search));
  }

  render() {
    const { loginState } = this.props;

    console.log('This is the login state :', loginState);
    return (
      <div>{loginState === constants.LOGIN_SUCCESS && <Redirect to="/" />}</div>
    );
  }
}

const mapStateToProps = ({ login: { loginState, errorMessage } }) => {
  return { loginState, errorMessage };
};

export default connect(mapStateToProps)(SocialAuth);
