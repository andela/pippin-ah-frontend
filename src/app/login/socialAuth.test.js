import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import TwitterLogin from 'react-twitter-auth';
import axios from 'axios';
import SocialAuth from './SocialAuth';
import { LoginComponent as Login } from './LoginComponent';

jest.mock('axios');

let props;

describe('SOCIAL AUTH TEST SUITE', () => {
  describe('SocialAuth Component', () => {
    it('should render the Social Auth Page', () => {
      const component = shallow(<SocialAuth />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('SocialAuth Component', () => {
    const initialState = {
      login: {
        loginState: '',
        errorMessage: '',
      },
    };
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);
    props = {
      dispatch: () => {},
      authLogin: () => {},
    };

    it('should render the Social Auth Page with google sing in', () => {
      global.window = Object.create(window);
      const url = '?google&code=4/_gDuguR9sgdfg';
      Object.defineProperty(window, 'location', {
        value: {
          href: url,
          search: url,
        },
      });
      const response = {
        data: {
          username: 'Diamond',
          email: 'valsido@gmail.com',
          token: 'cardinalsajdlsfslsfkjdlsfjflsfj',
        },
      };

      axios.get.mockResolvedValue(response);
      const component = mount(
        <Provider store={store}>
          <SocialAuth {...props} />
        </Provider>,
      );
      expect(component).toMatchSnapshot();
    });

    it('should render the Social Auth Page with facebook sign in', () => {
      global.window = Object.create(window);
      const url = '?facebook&code=thisisthestory';
      Object.defineProperty(window, 'location', {
        value: {
          href: url,
          search: url,
        },
      });
      const response = {
        data: {
          username: 'Diamond',
          email: 'valsido@gmail.com',
          token: 'cardinalsajdlsfslsfkjdlsfjflsfj',
        },
      };

      axios.get.mockResolvedValue(response);
      const component = mount(
        <Provider store={store}>
          <SocialAuth {...props} />
        </Provider>,
      );
      expect(component).toMatchSnapshot();
    });

    it('should throw exception when authentication fails', () => {
      global.window = Object.create(window);
      const url = '?facebook&code=thisisthestory';
      Object.defineProperty(window, 'location', {
        value: {
          href: url,
          search: url,
        },
      });
      const response = {
        data: {
          username: 'Diamond',
          email: 'valsido@gmail.com',
          token: 'cardinalsajdlsfslsfkjdlsfjflsfj',
        },
      };

      axios.get.mockResolvedValue(response);
      const component = mount(
        <Provider store={store}>
          <SocialAuth {...props} />
        </Provider>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
