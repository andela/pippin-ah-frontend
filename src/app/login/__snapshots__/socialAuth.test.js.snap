// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SOCIAL AUTH TEST SUITE SocialAuth Component should render the Social Auth Page 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`SOCIAL AUTH TEST SUITE SocialAuth Component should render the Social Auth Page with facebook sign in 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SocialAuth)
    authLogin={[Function]}
    dispatch={[Function]}
  >
    <SocialAuth
      authLogin={[Function]}
      dispatch={[Function]}
      errorMessage=""
      loginState=""
    >
      <div
        className="centerDiv"
      >
        <div>
          <RingLoaderComponent>
            <div
              className="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </RingLoaderComponent>
          <p>
            Logging in...
          </p>
        </div>
      </div>
      <div />
    </SocialAuth>
  </Connect(SocialAuth)>
</Provider>
`;

exports[`SOCIAL AUTH TEST SUITE SocialAuth Component should render the Social Auth Page with google sing in 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SocialAuth)
    authLogin={[Function]}
    dispatch={[Function]}
  >
    <SocialAuth
      authLogin={[Function]}
      dispatch={[Function]}
      errorMessage=""
      loginState=""
    >
      <div
        className="centerDiv"
      >
        <div>
          <RingLoaderComponent>
            <div
              className="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </RingLoaderComponent>
          <p>
            Logging in...
          </p>
        </div>
      </div>
      <div />
    </SocialAuth>
  </Connect(SocialAuth)>
</Provider>
`;

exports[`SOCIAL AUTH TEST SUITE SocialAuth Component should throw exception when authentication fails 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SocialAuth)
    authLogin={[Function]}
    dispatch={[Function]}
  >
    <SocialAuth
      authLogin={[Function]}
      dispatch={[Function]}
      errorMessage=""
      loginState=""
    >
      <div
        className="centerDiv"
      >
        <div>
          <RingLoaderComponent>
            <div
              className="lds-ring"
            >
              <div />
              <div />
              <div />
              <div />
              <div />
            </div>
          </RingLoaderComponent>
          <p>
            Logging in...
          </p>
        </div>
      </div>
      <div />
    </SocialAuth>
  </Connect(SocialAuth)>
</Provider>
`;
